network-config-analyzer/ConnectionSet.py

```diff
@@ -88,13 +94,65 @@ def _get_connections_representation(self, is_str):
             res += self._get_protocols_ranges_representation(is_str, protocols_ranges)
         return ','.join(s for s in res) if is_str else res
 
+    @staticmethod
+    def _aggregate_connection_representation(protocols):
+        """
+        Aggregate shared properties of the protocols, for better human understanding.
+        :param dict protocols: a map from protocol number (1-255) to allowed properties
+        :return: dict protocols_not_aggregated: the rest of the protocol data that was not aggregated.
+        :return: str aggregation_results: a string of the aggregated representation
+        """
+        protocols_not_aggregated = protocols.copy()
```

Comment:
Can we delay copying the data structure (may be expensive) until we are sure we'll need a copy?